phpBB/phpbb/notification/type/mention.php

```diff
@@ -0,0 +1,397 @@
+	*/
+	public function get_email_template()
+	{
+		return 'topic_notify';
+	}
+
+	/**
+	* Get email template variables
+	*
+	* @return array
+	*/
+	public function get_email_template_variables()
+	{
+		if ($this->get_data('post_username'))
+		{
+			$username = $this->get_data('post_username');
+		}
+		else
+		{
+					$username = $this->user_loader->get_username($this->get_data('poster_id'), 'username');
+		}
+
+		return array(
+			'AUTHOR_NAME'				=> htmlspecialchars_decode($username),
+			'POST_SUBJECT'				=> htmlspecialchars_decode(censor_text($this->get_data('post_subject'))),
+			'TOPIC_TITLE'				=> htmlspecialchars_decode(censor_text($this->get_data('topic_title'))),
+
+			'U_VIEW_POST'				=> generate_board_url() . "/viewtopic.{$this->php_ext}?p={$this->item_id}#p{$this->item_id}",
+			'U_NEWEST_POST'				=> generate_board_url() . "/viewtopic.{$this->php_ext}?f={$this->get_data('forum_id')}&t={$this->item_parent_id}&e=1&view=unread#unread",
+			'U_TOPIC'					=> generate_board_url() . "/viewtopic.{$this->php_ext}?f={$this->get_data('forum_id')}&t={$this->item_parent_id}",
+			'U_VIEW_TOPIC'				=> generate_board_url() . "/viewtopic.{$this->php_ext}?f={$this->get_data('forum_id')}&t={$this->item_parent_id}",
+			'U_FORUM'					=> generate_board_url() . "/viewforum.{$this->php_ext}?f={$this->get_data('forum_id')}",
+			'U_STOP_WATCHING_TOPIC'		=> generate_board_url() . "/viewtopic.{$this->php_ext}?uid={$this->user_id}&f={$this->get_data('forum_id')}&t={$this->item_parent_id}&unwatch=topic",
+		);
+	}
+
+		/**
+	* {@inheritdoc}
+	*/
+	public function create_insert_array($post, $pre_create_data = array())
+	{
+		$this->set_data('poster_id', $post['poster_id']);
+
+		$this->set_data('topic_title', $post['topic_title']);
+
+		$this->set_data('post_subject', $post['post_subject']);
+
+		$this->set_data('post_username', (($post['poster_id'] == ANONYMOUS) ? $post['post_username'] : ''));
+
+		$this->set_data('forum_id', $post['forum_id']);
+
+		$this->set_data('forum_name', $post['forum_name']);
+
+		$this->notification_time = $post['post_time'];
+
+		// Topics can be "read" before they are public (while awaiting approval).
+		// Make sure that if the user has read the topic, it's marked as read in the notification
+		if ($this->inherit_read_status && isset($pre_create_data[$this->user_id]) && $pre_create_data[$this->user_id] >= $this->notification_time)
+		{
+			$this->notification_read = true;
+		}
+
+		parent::create_insert_array($post, $pre_create_data);
+	}
+
+	/**
+	* Get the HTML formatted title of this notification
+	*
+	* @return string
+	*/
+	public function get_title()
+	{
+		$responders = $this->get_data('responders');
+		$usernames = array();
+
+		if (!is_array($responders))
+		{
+			$responders = array();
+		}
+
+		$responders = array_merge(array(array(
+			'poster_id'		=> $this->get_data('poster_id'),
+			'username'		=> $this->get_data('post_username'),
+		)), $responders);
+
+		$responders_cnt = sizeof($responders);
```

Comment:
Please use ```count()``` instead as ```sizeof()``` can be sheduled for deprecation in PHP 7.2+ (see https://wiki.php.net/rfc/deprecations_php_7_2#suggested_deprecations).